activerecord/lib/active_record/persistence.rb

```diff
@@ -308,17 +308,23 @@ def instantiate(attributes, column_types = {}, &block)
       # for updating all records in a single query.
       def update(id = :all, attributes)
         if id.is_a?(Array)
+          if id.any? { |one_id| one_id.is_a?(ActiveRecord::Base) }
+            raise ArgumentError,
+              "You are passing an array of ActiveRecord::Base instances to `update`. " \
+              "Please pass the id of the objects by calling `pluck(:id)` or `map(&:id)`."
+          end
           id.map { |one_id| find(one_id) }.each_with_index { |object, idx|
             object.update(attributes[idx])
           }
         elsif id == :all
           all.each { |record| record.update(attributes) }
         else
-          if ActiveRecord::Base === id
+          if id.is_a?(ActiveRecord::Base)
```

Comment:
I don't think we should touch this line if we're not changing its behaviour.

```suggestion
          if ActiveRecord::Base === id
```